gen/enum_test.go

```diff
@@ -501,6 +502,69 @@ func TestEnumLabelValid(t *testing.T) {
 	}
 }
 
+func TestEnumStrictMarshalText(t *testing.T) {
+	tests := []struct {
+		testName   string
+		enumStrict tems.EnumMarshalStrict
+		hasError   bool
+	}{
+		{"marshalText success",
+			1,
+			false,
+		},
+		{
+			"marshalText unknown value",
+			5,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(string(tt.testName), func(t *testing.T) {
+			marshalledValue, err := tt.enumStrict.MarshalText()
+			if tt.hasError {
```

Comment:
Instead of hasError which replicates the logic for generating an error message, would you mind changing this to a hard-coded string in the test cases that specifies an approximate or exact error message?

Something like:

```
tests := []struct{
  desc string
  give tems.EnumMarshalStrict
  
  want    string // expected MarshalText value
  wantErr string // error message if any
}{

    // ...
}
```

Where only one of `want` and `wantErr` is set per test case. In the test logic itself, we can do something like:

```
got, err := tt.give.MarshalText()
if len(tt.wantErr) > 0 {
    require.Error(t, err)
    assert.Equal(t, tt.wantErr, err.Error())
} else {
    require.NoError(t, err)
    assert.Equal(t, tt.want, string(got))
}
```

This will allow us to validate all cases properly.

And I want to clarify that for `wantErr`, we should *hard-code* the exact string we want to see in the error message. Otherwise we're just replicating the code we wrote in the template, which doesn't verify that the output is legible. For example, if there's a mistake in the fmt.Sprintf template, then both, the test and code being tested will agree, but it won't actually catch the issue.

So for example, something like:

    wantErr: `unknown enum value "1" for "EnumFoo"`